src/virtualenv/seed/embed/wheels/periodic_update.py

```diff
@@ -0,0 +1,195 @@
+            yield update_log
+    except Timeout:
+        return
+
+
+class UpdateLog(object):
+    datetime_fmt = "%Y-%m-%dT%H:%M:%SZ"
+
+    def __init__(self, path):
+        content = {}
+        if path.exists():
+            try:
+                with open(str(path), "rt") as file_handler:
+                    content = json.load(file_handler)
+            except (IOError, ValueError):
+                pass
+        self.started = self._load_datetime(content, "started")
+        self.completed = self._load_datetime(content, "completed")
+
+        self.updated = content.get("updated")
+        self.updated_release_date = content.get("updated_release_date")
+
+        self.previous_updated = content.get("previous_updated")
+        self.previous_updated_release_date = content.get("previous_updated_release_date")
+        self.path = path
+
+    def _load_datetime(self, content, key):
+        value = content.get(key)
+        if value:
+            return datetime.strptime(value, self.datetime_fmt)
+        return None
+
+    def _dump_datetime(self, value):
+        if value is None:
+            return None
+        return datetime.strftime(value, self.datetime_fmt)
+
+    @property
+    def needs_update(self):
+        now = datetime.now()
+        if self.completed is None:  # never completed
+            return self._check_start(now)
+        else:
+            delta_since_last_completed = now - self.completed
+            if delta_since_last_completed < timedelta(days=13):
+                return False
+            return self._check_start(now)
+
+    def _check_start(self, now):
+        if self.started is None:  # never started
+            return True
+        delta_since_last_start = now - self.started
+        if delta_since_last_start >= timedelta(hours=1):  # over an hour ago -> operations crashed, try again
+            return True
+        return False
+
+    def update(self):
+        content = {
+            "started": self._dump_datetime(self.started),
+            "completed": self._dump_datetime(self.completed),
+            "updated": self.updated,
+            "updated_release_date": self._dump_datetime(self.updated_release_date),
+            "previous_updated": self.previous_updated,
+            "previous_updated_release_date": self._dump_datetime(self.previous_updated_release_date),
+        }
+        with open(str(self.path), "wt") as file_handler:
+            json.dump(content, file_handler, sort_keys=True, indent=4)
+
+
+def trigger_update(distribution, for_py_version, wheel, app_data):
+    cmd = [
+        sys.executable,
+        "-c",
+        "from virtualenv.seed.embed.wheels.periodic_update import do_update;"
+        "do_update({!r}, {!r}, {!r}, {!r})".format(distribution, for_py_version, str(wheel.path), str(app_data.path)),
+    ]
+    kwargs = {"stdout": subprocess.PIPE, "stderr": subprocess.PIPE}
+    if sys.platform == "win32":
+        kwargs["creation_flags"] = DETACHED_PROCESS
```

Comment:
The update is a detached new process to be run in the background (to not slow down creation).